### ISSUE TITLE
remove inappropriate transitive dependencies by splitting `conformance` into a separate module

### ISSUE THREAD

**Issue by bcmills** [golang/protobuf#806]
https://github.com/golang/protobuf/issues/804 reports an error with `go get -u` for this module.

The problem arises through a large transitive dependency graph, introduced via a dependency on `google.golang.org/genproto` (and eventually `google.golang.org/grpc` and its wide net of dependencies) in the `conformance/internal/conformance_proto` package:

```
~/src/github.com/golang/protobuf$ go mod why -m google.golang.org/genproto
# google.golang.org/genproto
github.com/golang/protobuf/conformance/internal/conformance_proto
google.golang.org/genproto/protobuf/field_mask
```

However, the `conformance` subtree doesn't even build using `cmd/go`: it has its own `Makefile` and `.sh` test driver. It should not be included in the main module, and then the inappropriate `grpc` dependency will be removed.

That will also help to address https://github.com/golang/lint/issues/436.

CC @dmitshur @thepudds

**Comment by dsnet** [comment]
We should just delete the conformance test. It hasn't been run in forever, and most of the new shiny stuff is in v2 anyways.

**Comment by dsnet** [comment]
Back-tracking a bit, it seems that the real issue is that `field_mask` is in `genproto`. It's entirely reasonable that this repo needs to depend on `field_mask` again in the future since it's a well-known type with special semantic meaning.

Thus, we should really move `field_mask` over to this repo.

**Comment by dsnet** [comment]
Related issues with `field_mask` are #745 and #398.

**Comment by bcmills** [comment]
That's fair, but I think for now let's stop the bleeding by cutting out the `conformance` module. Then we can do the rest of the cleanup as time permits.

**Comment by dsnet** [comment]
Fixed by #808.
